pkgs/build-support/rust/hooks/cargo-setup-hook.sh

```diff
@@ -36,6 +36,13 @@ cargoSetupPostUnpackHook() {
     @cargoConfig@
 EOF
 
+    local buildTop="/build-top"
+    if [ -n "$NIX_BUILD_TOP" ]; then
+      buildTop="$NIX_BUILD_TOP"
+    fi
+
+    substituteInPlace .cargo/config.toml --subst-var-by BUILD_TOP "$buildTop"
```

Comment:
This is bash, so this can be simplified to just using a variable substitution modifier

```suggestion
    substituteInPlace .cargo/config.toml --subst-var-by BUILD_TOP "${NIX_BUILD_TOP:-/build-top}"
```